client/src/features/userSlice.jsx

```diff
@@ -9,19 +9,39 @@ export const userSlice = createSlice({
             userName: 'Frasia',
             role: 'admin',
             email: 'frasia.nyakundi@student.moringaschool.com'
-        }
+        },
+        users: []
     },
     reducers: {
         login: (state, action) => {
             state.user = action.payload;
         },
         logout: (state) => {
             state.user = null
+        },
+        fetchUsers: (state, action) => {
+            state.users = action.payload;
+        },
+        extraReducers: (builder) => {
+            builder.addCase(getUsers.fulfilled, (state, action) => {
+                state.users = action.payload
+            })
         }
     }
+
 })
+export const getUsers = createAsyncThunk(
+    'user/getUsers',
+    async() => {
+        const response = await fetch (
+            "https://myduka-group-5-mnnj.onrender.com/users"
```

Comment:
I can see this URL is pointing directly to the deployed server, this will affect your database when doing development. Consider using the environment variables for the base URL and then hard-code the endpoint.